Week3/hanoi/src/Hanoi.java

```diff
@@ -1,70 +1,54 @@
-import java.util.Stack;
-
+// Set bars to execute hanoi algorithm.
 public class Hanoi {
-  int num_disks;
-
-  Stack<Integer> bar1 = new Stack<>();
 
-  Stack<Integer> bar2 = new Stack<>();
+  int numDisks = 0;
 
-  Stack<Integer> bar3 = new Stack<>();
+  // Initialize the numDisks and put disks in bar1 from larger disks to smaller disks.
+  // Numbers in the bar mean the size of a disk.
+  public Hanoi(int numDisks) {
+    this.numDisks = numDisks;
+  }
 
-  public Hanoi (int num_disks) {
-    this.num_disks = num_disks;
-    
-    for (int i=num_disks; i>=1; i--) {
-      bar1.push(i);
-    }
+  // Print log of disk movement.
+  private void moveDisk(int diskNumber, int fromBar, int toBar) {
+    System.out.println("Disk "+diskNumber+" moved from bar "+fromBar+" to bar "+toBar);
   }
 
-  public int hanoi (int num_disks, int from_bar, int second_bar, int to_bar) {
-    if (num_disks == 1) {
+  // Returns the answer of Hanoi problem with initialized numDisks
+  public int calculateHanoiSteps() throws HanoiInputException {
+    return calculateHanoiSteps(numDisks, 1, 2, 3);
+  }
 
-      /* Pop a disk from from_bar */
-      int disk_toMove = 0;
+  // Returns the answer of Hanoi problem with given numDisks.
+  // Prints disk movement log.
+  public int calculateHanoiSteps(int numDisks, int fromBar, 
```

Comment:
You implement the two function of cacluateHanoiSteps(). It's meaningless.
You should focus on solving Hanoi with three bar.
Also please think about meaning of int fromBar. It's just a name of bar.

Reply:
I deleted calculateHanoiSteps which does not need arguments.